stream/stream.go

```diff
@@ -43,24 +42,34 @@ func (i ID) String() string {
 	return string(i)
 }
 
-func Record[T any](stream T, events *Events[T], e Event[T]) error {
+func Record[T Stream[T]](stream *T, e Event[T]) error {
 	err := e.ApplyTo(stream)
 	if err != nil {
 		return err
 	}
 
-	events.Record(e)
+	(*stream).Events().Record(e)
 
 	return nil
 }
 
-func ApplyAll[T Stream[T]](stream T) error {
-	for _, e := range stream.PopEvents() {
-		err := e.ApplyTo(stream)
+func New[T Stream[T]](eventsSlice []VersionedEvent[T]) (T, error) {
```

Comment:
I suppose `New` should also return `*T`?

Reply:
And also Save to use `*T` I believe

Reply:
https://github.com/ThreeDotsLabs/esja/pull/14/commits/8877732c832607fb7329105b0ce47d12c0417a0b